Blockstack/Classes/GaiaHubSession.swift

```diff
@@ -283,12 +283,16 @@ class GaiaHubSession {
             getReadURL.then({ url in
                 let task = URLSession.shared.dataTask(with: url) { data, response, error in
                     guard error == nil,
+                        let httpResponse = response as? HTTPURLResponse,
                         let data = data else {
                             print("Gaia hub store request error")
                             reject(GaiaError.requestError)
                             return
                     }
-                    let contentType = (response as? HTTPURLResponse)?.allHeaderFields["Content-Type"] as? String ?? "application/json"
+                    if httpResponse.statusCode == 404 {
```

Comment:
Should we only resolve if `httpResponse.statusCode == 200`, and reject otherwise?
If not, we probably want to handle [all the possible errors](https://github.com/blockstack/gaia/blob/a723e8ce2af581c1c2a28aecd564ed7d46c0b3bf/hub/src/server/http.ts#L62):
```js
        if (err instanceof errors.ValidationError) {
          writeResponse(res, { message: err.message, error: err.name }, 401)
        } else if (err instanceof errors.AuthTokenTimestampValidationError) {
          writeResponse(res, { message: err.message, error: err.name  }, 401)
        } else if (err instanceof errors.BadPathError) {
          writeResponse(res, { message: err.message, error: err.name  }, 403)
        } else if (err instanceof errors.NotEnoughProofError) {
          writeResponse(res, { message: err.message, error: err.name  }, 402)
        } else if (err instanceof errors.ConflictError) {
          writeResponse(res, { message: err.message, error: err.name  }, 409)
        } else {
          writeResponse(res, { message: 'Server Error' }, 500)
        }
```

Reply:
I've modified it to allow only 200s, return the `fileNotFoundError` for 404, and a `serverError` for anything else. We probably do need more descriptive errors in iOS (beyond just the `GaiaError` enum) but I think that would be beyond the scope of this PR.